src/init.cpp

```diff
@@ -384,6 +384,7 @@ void DynamicDataLoader::initialize()
 
     add( "tool_quality", &quality::load_static );
     add( "technique", &load_technique );
+    add( "weapon_category", &weapon_category::load_weapon_categories );
```

Comment:
This probably should be unloaded in `DynamicDataLoader::unload_data`.

Reply:
Is there any kind of rule I can follow to know when I should add to `DynamicDataLoader::unload_data`? For example `speed_description` and `mood_face` which I've added didn't go into `DynamicDataLoader::unload_data` ( mostly because I didn't know about it ).

Reply:
Basically, `unload_data` should undo what `initialize` has done when loading a save. Otherwise the loaded objects are leaked to the next loaded save, and the save may behave incorrectly if different objects are defined for it (via a different mod, for example).